core/iwasm/aot/aot_runtime.c

```diff
@@ -39,8 +39,10 @@ bh_static_assert(offsetof(AOTModuleInstance, func_type_indexes)
                  == 6 * sizeof(uint64));
 bh_static_assert(offsetof(AOTModuleInstance, cur_exception)
                  == 13 * sizeof(uint64));
-bh_static_assert(offsetof(AOTModuleInstance, global_table_data)
+bh_static_assert(offsetof(AOTModuleInstance, aot_stack_sizes)
                  == 13 * sizeof(uint64) + 128 + 11 * sizeof(uint64));
+bh_static_assert(offsetof(AOTModuleInstance, global_table_data)
+                 == 13 * sizeof(uint64) + 128 + 12 * sizeof(uint64));
```

Comment:
i'm not happy to break offsets.
should i add the new member after the variable parts of the structure?

Reply:
Adding new fields in the WASMModuleInstance/AOTModuleInstance may cause compatibility issue: the AOT code also use pre-calculated offsets to access the global data and table instance (which are at the end of AOTModuleInstance and are allocated together with AOTModuleInstance), these offsets will change if new field is inserted, and the old AOT file (generated by older wamrc) won't run well with the new runtime.

How about adding the `aot_stack_sizes` into exec_env, after field `native_stack_top_min`:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/common/wasm_exec_env.h#L91

Reply:
is there a longer term plan about compatibility breakage?
like having a flag day for https://github.com/bytecodealliance/wasm-micro-runtime/issues/2144 when we can clean up these things?

Reply:
i moved it to aot inst extra.

Reply:
> is there a longer term plan about compatibility breakage? like having a flag day for #2144 when we can clean up these things?

Yes, we hope to upgrade the AOT file format in the GC AOT feature and don't support the old AOT file format again after the feature is implemented. The GC AOT is supposed to be finished in this year, and we may merge it into main branch sometime this year.